samples/mqtt/basic_pub_sub/main.cpp

```diff
@@ -238,6 +238,12 @@ int main(int argc, char *argv[])
             std::lock_guard<std::mutex> lockGuard(mutex);
             connectionSucceeded = false;
         }
+        else if (returnCode != AWS_MQTT_CONNECT_ACCEPTED)
```

Comment:
Even better idea:

in aws-c-mqtt, if the returnCode is non-zero then error_code should be non-zero.

We'd need to add a new aws-error like: `AWS_ERROR_MQTT_CONNECTION_REFUSED`

Here are the docs for return code: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc385349257

They're all just specific reasons for a failed connection.

> If a server sends a CONNACK packet containing a non-zero return code it MUST then close the Network Connection [MQTT-3.2.2-5].

Reply:
That makes sense.  Maybe I can trick Dengke into doing it.